src/instruction/instructions/adcs.ts

```diff
@@ -0,0 +1,59 @@
+import { Halfword, Word } from 'types/binary'
+
+import { AluResult, add } from 'board/alu'
+import { IMemory } from 'board/memory/interfaces'
+import { Flag, Registers } from 'board/registers'
+
+import { ILabelOffsets } from 'instruction/interfaces'
+import {
+  checkOptionCount,
+  create,
+  createLowRegisterBits,
+  getBits,
+  setBits
+} from 'instruction/opcode'
+
+import { BaseInstruction } from './base'
+
+export class AdcsInstruction extends BaseInstruction {
+  public name: string = 'ADCS'
+  public pattern: string = '0100000101XXXXXX'
+  private rdnPattern: string = '0100000101000XXX'
+  private rmPattern: string = '0100000101XXX000'
+
+  public encodeInstruction(options: string[], labels: ILabelOffsets): Halfword {
+    checkOptionCount(options, 2, 3)
+    if (options.length === 3 && options[0] !== options[1]) {
+      throw new Error(
+        'First and second parameter must be the same register (Rdn = Rdn + Rm + C).'
+      )
```

Comment:
The user has influence by writing code wheter option 1 and 2 are the same or not when 3 options are provided. IMO here VirtualBoardError would make sense

Reply:
I think right now having the message in a standard `Error` instance is good enough. The code editor anyways does not do something with this information apart from showing the message on the user interface. I would rather wait until we start to work on the `compile errors` story ([VCB-22](https://virtualctboard.atlassian.net/browse/VCB-22)).